src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -1385,6 +1389,41 @@ namespace
         }
         else
         {
+            asr::AOVContainer aovs;
+
+            IRenderElementMgr* re_manager = GetCOREInterface()->GetCurRenderElementMgr();
+            re_manager->SetDisplayElements(false);
+            for (size_t i = 0, e = re_manager->NumRenderElements(); i < e; ++i)
+            {
+                const asr::AOVFactoryRegistrar factory_registrar;
```

Comment:
Move the creation of the factory registrar out of the loop: creating a factory registrar is an extremely expensive operation as it will scan search paths for plugins, load DLLs, etc. In fact we might want to create the registrar once when the plugin is loaded rather than everytime a project is built.